nbqa/save_source.py

```diff
@@ -9,51 +9,40 @@
 from itertools import takewhile
 from typing import TYPE_CHECKING, Dict, Iterator, List, Optional, Tuple
 
+from nbqa.notebook_info import NotebookInfo
+
 if TYPE_CHECKING:
     from pathlib import Path
 
 CODE_SEPARATOR = "# %%"
 MAGIC = ["%%script", "%%bash"]
-IGNORE_CELL_REPLACEMENT = "# nbqa"
 IGNORE_LINE_REPLACEMENT = "pass  # nbqa"
 
 
-def _replace_magics(
-    source: List[str], ignore_cells: List[str]
-) -> Iterator[Tuple[str, Optional[str]]]:
+def _replace_line_magics(source: List[str]) -> Iterator[Tuple[str, Optional[str]]]:
     """
-    Comment out lines with magic commands.
-
-    If it's a script/bash cell, comment out the entire cell.
+    Replace IPython line magics with valid python code.
 
     Parameters
     ----------
     source
         Source from notebook cell.
-    ignore_cells
-        Extra cells which nbqa should ignore.
 
     Yields
     ------
     str
-        Line from cell, possibly commented out.
+        Lines from cell, with line magics replaced with python code
     """
-    ignore = MAGIC + [i.strip() for i in ignore_cells]
-
-    ignore_cell = source and any(source[0].lstrip().startswith(i) for i in ignore)
     for line in source:
         if (
-            line.lstrip().startswith("!") or line.lstrip().startswith("%")
-        ) or ignore_cell:
+            line.lstrip().startswith("!")
+            or line.lstrip().startswith("%")
+            or line.rstrip().endswith("?")
```

Comment:
Nice, hadn't thought of that!

Reply:
I think this can be refactored to:
```python 
if line.lstrip().startswith(("!", "%")) or line.rstrip().endswith("?"):
```
And since `?some_object` is also valid magic, it should be:
```python 
if line.lstrip().startswith(("!", "%", "?")) or line.rstrip().endswith("?"):
```

Reply:
> And since ?some_object is also valid magic,

there's no end to my surprises about Jupyter Notebooks